CHANGELOG.md

```diff
@@ -2,6 +2,14 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## [3.56.0](https://github.com/sonata-project/SonataAdminBundle/compare/3.55.0...3.56.0) - 2019-12-05
```

Comment:
I finally got around to writing the few lines of bash that allow to generate that line and insert it in the right place. Now my script is better:

```bash
#!/bin/bash -eu
function stable_branch()
{
	git branch --format='%(refname:lstrip=2)'|grep '^[0-9].x$'|sort|tail --lines=1
}

function latest_tag()
{
	git describe --abbrev=0 --tags
}

function latest_tag_date()
{
	git --no-pager show --format=%aI --no-patch "$(latest_tag)"|tail --lines 1
}

function urlencode()
{
	read -r toEncode
	php --run "echo urlencode('$toEncode');"
}

function current_branch()
{
	git rev-parse --abbrev-ref HEAD
}

function repo_path()
{
	git remote --verbose \
		| grep ':sonata' \
		| grep 'git@' \
		| head --lines=1 \
		| cut --delimiter=':' --fields=2 \
		| cut --delimiter='.' --fields=1
}

function today()
{
	date +%Y-%m-%d
}


function new_tag()
{
	echo "What type is that version number (minor/patch)" >&2
	read -rp "> " versionType
	IFS='.' read -r -a version <<<"$(latest_tag)"
	case "$versionType" in
		minor)
			((version[1]++))
			;;
		patch)
			((version[2]++))
			;;
		*)
			echo "wut?"
			exit 1
	esac
	echo "${version[0]}.${version[1]}.${version[2]}"
}

git checkout "$(stable_branch)"
git pull --all --prune
xdg-open "https://github.com/$(repo_path)/pulls?q=base:$(current_branch)+merged:>$(latest_tag_date|urlencode)"
newTag=$(new_tag)
sed -i "4a\
	## [$newTag](https://github.com/$(repo_path)/compare/$(latest_tag)...$newTag) - $(today)" CHANGELOG.md
$EDITOR CHANGELOG.md
grep --fixed-strings --recursive --line-number "$(stable_branch)" src tests # look for deprecations
```